tasks/main.yml

```diff
@@ -49,6 +49,7 @@
   register: __firewall_config_files_before
   changed_when: false
   when: __firewall_previous_replaced | bool
+  check_mode: false
```

Comment:
While this would prevent errors, I think disabling check mode for this task would be a bad idea if we want to support check mode with `previous: replaced`, or at least it is insufficient on its own.

In order to make this a full solution, I'd recommend modifying files/get_files_checksums.sh to support check mode when `remove=true`

Reply:
What should we report when the user runs this with check mode?

```yaml
- name: Run role
  include_role:
    name: linux-system-roles.firewall
  vars:
    firewall:
      - previous: replaced
```

That is - how do we know that the firewall configuration is already in the "default" state?